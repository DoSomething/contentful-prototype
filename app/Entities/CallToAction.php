<?php

namespace App\Entities;

use JsonSerializable;

class CallToAction extends Entity implements JsonSerializable
{
    /**
     * Convert the object into something JSON serializable.
     *
     * @return array
     */
    public function jsonSerialize()
    {
        switch ($this->getContentType()) {
            case 'callToAction':
                $type = $this->getContentType();
                $content = $this->content;
                $impactPrefix = $this->impactPrefix ?: null;
                $impactSuffix = $this->impactSuffix ?: null;
                $impactValue = $this->impactValue ?: null;
                break;

            case 'customBlock':
                $type = 'callToAction';
                $content = $this->title;
                $impactPrefix = $this->additionalContent['impactPrefix'] ?: null;
                $impactSuffix = $this->additionalContent['impactMessage'] ?: null;
                $impactValue = $this->additionalContent['impactNumber'] ?: null;
                break;
        }

        return [
            'id' => $this->entry->getId(),
            'type' => $type,
            'fields' => [
                'content' => $content,
                'displayOptions' =>$this->displayOptions->first(),
                'impactPrefix' => $impactPrefix,
                'impactSuffix' => $impactSuffix,
                'impactValue' => $impactValue,
                'useCoverImage' => $this->useCoverImage ?: null,
                'photo' => $this->photo ?: null,
            ],
        ];
    }
}
