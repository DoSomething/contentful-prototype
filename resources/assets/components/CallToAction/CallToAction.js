import React from 'react';
import gql from 'graphql-tag';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Card from '../utilities/Card/Card';
import { isScholarshipAffiliateReferral } from '../../helpers';
import { SCHOLARSHIP_SIGNUP_BUTTON_TEXT } from '../../constants';
import SignupButtonContainer from '../SignupButton/SignupButtonContainer';

export const CallToActionBlockFragment = gql`
  fragment CallToActionBlockFragment on CallToActionBlock {
    visualStyle
    useCampaignTagline
    content
    impactPrefix
    impactValue
    impactSuffix
    actionText
  }
`;

const renderImpactContent = (prefix, value, suffix) => {
  const valueElem = (
    <span className="block font-league-gothic font-normal text-4xl">
      {value}
    </span>
  );

  return (
    <div className="font-bold mb-6 uppercase">
      {prefix ? `${prefix} ` : null} {valueElem} {suffix ? ` ${suffix}` : null}
    </div>
  );
};

const CallToAction = ({
  className,
  content,
  impactPrefix,
  impactSuffix,
  impactValue,
  hideIfSignedUp,
  isSignedUp,
  tagline,
  useCampaignTagline,
  visualStyle,
}) => {
  if (hideIfSignedUp && isSignedUp) {
    return null;
  }

  return (
    <Card
      className={classnames('p-3 text-center', className, {
        'bg-white border border-solid border-gray-200 rounded':
          visualStyle === 'light',
        'bg-black-important rounded text-white': visualStyle === 'dark',
      })}
    >
      {useCampaignTagline ? (
        <div
          className={classnames('font-bold mb-6', {
            'text-yellow-500': visualStyle === 'dark',
          })}
        >
          {tagline}
        </div>
      ) : null}

      {impactValue
        ? renderImpactContent(impactPrefix, impactValue, impactSuffix)
        : null}

      {content ? (
        <div
          className={classnames('max-w-lg mb-6 mx-auto', {
            italic: visualStyle === 'light',
          })}
        >
          {content}
        </div>
      ) : null}

      {isSignedUp ? null : (
        <SignupButtonContainer
          text={
            isScholarshipAffiliateReferral()
              ? SCHOLARSHIP_SIGNUP_BUTTON_TEXT
              : undefined
          }
        />
      )}
    </Card>
  );
};

CallToAction.propTypes = {
  className: PropTypes.string,
  content: PropTypes.string,
  impactPrefix: PropTypes.string,
  impactSuffix: PropTypes.string,
  impactValue: PropTypes.string,
  hideIfSignedUp: PropTypes.bool,
  isSignedUp: PropTypes.bool.isRequired,
  tagline: PropTypes.string.isRequired,
  useCampaignTagline: PropTypes.bool,
  visualStyle: PropTypes.string,
};

CallToAction.defaultProps = {
  className: null,
  content: null,
  impactPrefix: null,
  impactSuffix: null,
  impactValue: null,
  hideIfSignedUp: false,
  useCampaignTagline: false,
  visualStyle: null,
};

export default CallToAction;
