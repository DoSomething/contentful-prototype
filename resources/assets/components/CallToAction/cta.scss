@import '../../scss/next-toolbox';

.call-to-action {
  background-color: $white;
  border-top: 1px solid $primary-border-color;
  padding: $half-spacing;

  &.-sticky {
    bottom: -1px;
    position: sticky;
    z-index: $sticky-cta-zindex;

    @include media($medium) {
      display: none;
    }

    > button {
      width: 100%;
    }
  }
}
