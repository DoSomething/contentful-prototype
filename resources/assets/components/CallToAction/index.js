export CallToActionContainer from './CallToActionContainer';

export default from './CallToAction';
