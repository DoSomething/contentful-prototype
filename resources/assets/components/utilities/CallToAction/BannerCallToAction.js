import { React } from 'react';
import PropTypes from 'prop-types';
import { snakeCase } from 'lodash';
import classnames from 'classnames';

import AnalyticsWaypoint from '../AnalyticsWaypoint/AnalyticsWaypoint';

const BannerCallToAction = ({
  attributes,
  children,
  colorClasses,
  message,
  stacked,
  title,
  waypointName,
}) => {
  return (
    <article
      className={classnames(
        'base-12-grid py-16 relative',
        colorClasses.background || 'bg-yellow-500',
      )}
      {...attributes}
    >
      <AnalyticsWaypoint
        className="absolute top-0 w-full"
        name={snakeCase(`${waypointName} top`)}
      />

      <div
        className={classnames('grid-wide text-center', {
          'xl:flex xl:items-center': !stacked,
        })}
      >
        <div className={classnames({ 'text-left xl:w-8/12': !stacked })}>
          <h1
            className={classnames(
              'font-bold text-2xl',
              colorClasses.text || 'text-gray-900',
            )}
          >
            {title}
          </h1>

          <p
            className={classnames(
              'text-lg',
              colorClasses.text || 'text-gray-900',
            )}
          >
            {message}
          </p>
        </div>

        <div className={classnames({ 'flex-grow': !stacked })}>{children}</div>
      </div>

      <AnalyticsWaypoint
        className="absolute bottom-0 w-full"
        name={snakeCase(`${waypointName} bottom`)}
      />
    </article>
  );
};

BannerCallToAction.propTypes = {
  attributes: PropTypes.object,
  children: PropTypes.node.isRequired,
  colorClasses: PropTypes.shape({
    background: PropTypes.string,
    text: PropTypes.string,
  }),
  message: PropTypes.string.isRequired,
  stacked: PropTypes.bool,
  title: PropTypes.string.isRequired,
  waypointName: PropTypes.string,
};

BannerCallToAction.defaultProps = {
  attributes: {},
  colorClasses: {},
  stacked: false,
  waypointName: 'banner cta',
};

export default BannerCallToAction;
