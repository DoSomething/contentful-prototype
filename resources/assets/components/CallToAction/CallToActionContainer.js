import { connect } from 'react-redux';

import CallToAction from './CallToAction';
import { isSignedUp } from '../../selectors/signup';

/**
 * Provide state from the Redux store as props for this component.
 */
const mapStateToProps = state => ({
  campaignId: state.campaign.id,
  coverImageUrl: state.campaign.coverImage.url,
  isSignedUp: isSignedUp(state),
  legacyCampaignId: state.campaign.legacyCampaignId,
  tagline: state.campaign.callToAction,
});

/**
 * Export the container component.
 */
export default connect(mapStateToProps)(CallToAction);
