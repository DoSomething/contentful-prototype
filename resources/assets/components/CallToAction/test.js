import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CallToAction from './CallToAction';

const component = shallow(<CallToAction className="sample-class" onClick={() => {}} />)

test('CallToAction snapshot test', () => {
  const tree = shallowToJson(component);

  expect(tree).toMatchSnapshot();
});
