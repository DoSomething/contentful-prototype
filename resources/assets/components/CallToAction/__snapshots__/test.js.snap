// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallToAction snapshot test 1`] = `
<div
  className="call-to-action sample-class"
/>
`;
